env0/resource_environment.go

```diff
@@ -0,0 +1,184 @@
+				Type:        schema.TypeString,
+				Description: "the environment's id",
+				Optional:    true,
+				Computed:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The environment's name",
+				Required:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "the environment's project id",
+				Required:    true,
+			},
+			"template_id": {
+				Type:        schema.TypeString,
+				Description: "the environment's template id",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func setEnvironmentSchema(d *schema.ResourceData, environment client.Environment) {
+	d.Set("name", environment.Name)
+	d.Set("project_id", environment.ProjectId)
+	d.Set("template_id", environment.TemplateId)
+}
+
+func resourceEnvironmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	payload := getCreatePayload(d)
+
+	environment, err := apiClient.EnvironmentCreate(payload)
+	if err != nil {
+		return diag.Errorf("could not create environment: %v", err)
+	}
+
+	d.SetId(environment.Id)
+	setEnvironmentSchema(d, environment)
+
+	return nil
+}
+
+func resourceEnvironmentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	environment, err := apiClient.Environment(d.Id())
+	if err != nil {
+		return diag.Errorf("could not get environment: %v", err)
+	}
+
+	setEnvironmentSchema(d, environment)
+
+	return nil
+}
+
+func resourceEnvironmentUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	payload := getUpdatePayload(d)
+
+	// TODO: deploy if needed
+
+	_, err := apiClient.EnvironmentUpdate(d.Id(), payload)
+	if err != nil {
+		return diag.Errorf("could not update environment: %v", err)
+	}
+
+	return nil
+}
+
+func resourceEnvironmentDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	_, err := apiClient.EnvironmentDestroy(d.Id())
+	if err != nil {
+		return diag.Errorf("could not delete team: %v", err)
+	}
+	return nil
+}
+
+func getCreatePayload(d *schema.ResourceData) client.EnvironmentCreate {
+	payload := client.EnvironmentCreate{}
+
+	if name, ok := d.GetOk("name"); ok {
+		payload.Name = name.(string)
+	}
+
+	if projectId, ok := d.GetOk("project_id"); ok {
+		payload.ProjectId = projectId.(string)
+	}
+
+	payload.DeployRequest = &client.DeployRequest{}
+
+	if blueprintId, ok := d.GetOk("template_id"); ok {
+		payload.DeployRequest.BlueprintId = blueprintId.(string)
+	}
+
+	if blueprintRepository, ok := d.GetOk("repository"); ok {
+		payload.DeployRequest.BlueprintRepository = blueprintRepository.(string)
+	}
+
+	if blueprintRevision, ok := d.GetOk("revision"); ok {
+		payload.DeployRequest.BlueprintRepository = blueprintRevision.(string)
```

Comment:
`BlueprintRepository` -> `BlueprintRevision`/`Revision`?

Reply:
the payload type corresponds to [our api](https://developer.env0.com/docs/api/b3A6MTkzNDg1MTk-create-environment) 
do you want me to change the field name on the resource? ("revision")

Reply:
My comment was referring to the typo you had here basically, but the "revision" attribute should be the same as saved on the environment/deployment log model, which I think is "blueprint revision"

Reply:
Oh I misunderstood 76c4908
and regarding the the field name on the resource I just went with what the ticket suggested (i.e field names that are not 1 to 1 with our api )